### PR TITLE
Only log errors if secrets in the manifest are missing the NextRotationOn tag

### PR DESCRIPTION
**Pull request by dkurepa**
https://github.com/dotnet/arcade/issues/12792

The NextRotationOn property is already in the Tags, we don't need to have a separate Property just for it, since it's used only in one place

### CODE REVIEW

File: src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs [hunk 1]
```diff
@@ -84,6 +85,12 @@ public override async Task RunAsync(CancellationToken cancellationToken)
             foreach (var (name, secret, secretType, secretReferences) in orderedSecretTypes)
             {
                 _console.WriteLine($"Synchronizing secret {name}, type {secret.Type}");
+
+                if (!existingSecrets[name].Tags.ContainsKey(AzureKeyVault.NextRotationOnTag))
```
Comment (premun) [inline src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs:89]: Just asking as I don't know the code much:
What if the secret is not in the KeyVault? Is it possible? Will this throw?

File: src/Microsoft.DncEng.SecretManager/StorageTypes/AzureKeyVault.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ public class AzureKeyVaultParameters
 [Name("azure-key-vault")]
 public class AzureKeyVault : StorageLocationType<AzureKeyVaultParameters>
 {
-    private static readonly string _nextRotationOnTag = "next-rotation-on";
+    public static readonly string NextRotationOnTag = "next-rotation-on";
```
Comment (premun) [inline src/Microsoft.DncEng.SecretManager/StorageTypes/AzureKeyVault.cs:24]:
```suggestion
    public const string NextRotationOnTag = "next-rotation-on";
```

File: src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs (hunk 1 above)
Comment (dkurepa) [reply src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs]: I rewrote that part so it doesn't throw.
Later in the code, we go through all secrets that are in the manifest but not in the Vault and create them, so we shouldn't be doing anything here if we don't find the secret

File: src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs [hunk 3]
```diff
@@ -84,6 +85,15 @@ public override async Task RunAsync(CancellationToken cancellationToken)
             foreach (var (name, secret, secretType, secretReferences) in orderedSecretTypes)
             {
                 _console.WriteLine($"Synchronizing secret {name}, type {secret.Type}");
+
+                if (existingSecrets.TryGetValue(name, out var existingSecret))
+                {
+                    if (!existingSecret.Tags.ContainsKey(AzureKeyVault.NextRotationOnTag))
```
Comment (premun) [inline src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs:91]: I'd merge these ifs, but it's a personal preference

File: src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs [hunk 4]
```diff
@@ -84,6 +85,15 @@ public override async Task RunAsync(CancellationToken cancellationToken)
...
+                    {
+                        _console.LogError($"Key Vault Secret '{name}' is missing {AzureKeyVault.NextRotationOnTag} tag, using the end of time as value. Please force a rotation or manually set this value.");
```
Comment (premun) [inline src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs:93]: Does the `using the end of time as value` part still apply?

Comment (dkurepa) [reply src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs:93]: it does, later we're going through the SecretProperties and using the end of time value for those that don't have the `NextRotationOnTag`

File: src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs (hunk 3 above)
Comment (dkurepa) [reply src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs]: done

File: src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs (hunk 4 above)
Comment (dkurepa) [reply src/Microsoft.DncEng.SecretManager/Commands/SynchronizeCommand.cs]: True, I improved the logging a bit

**Review by premun — DISMISSED** [review]

**Review by premun — APPROVED** [review]

**Pull request merged**